samza-test/src/main/java/org/apache/samza/test/framework/TestRunner.java

```diff
@@ -148,28 +158,30 @@ public static TestRunner of(StreamApplication app) {
   }
 
   /**
-   * Only adds a config from {@code config} to samza job {@code configs} if they dont exist in it.
-   * @param config configs for the application
+   * Adds a config to Samza application. This config takes precedence over default configs and descriptor generated configs
+   *
+   * @param key of the config
+   * @param value of the config
    * @return this {@link TestRunner}
    */
-  public TestRunner addConfigs(Map<String, String> config) {
-    Preconditions.checkNotNull(config);
-    config.forEach(this.configs::putIfAbsent);
+  public TestRunner addConfig(String key, String value) {
+    Preconditions.checkNotNull(key);
+    Preconditions.checkNotNull(value);
+    String configKeyPrefix = String.format(JobConfig.CONFIG_JOB_PREFIX(), JOB_NAME);
```

Comment:
Please refer to PR #642 , we need to have the full jobNameAndId in the config job prefix.

Reply:
Did that